backend/api/routes/files.py

```diff
@@ -217,11 +241,19 @@ async def update_file(
     session.execute(stmt)
 
 
-@router.delete("/{project_id}/files/{file_id}", status_code=codes.NO_CONTENT)
+@router.delete("/{project_id}/files/{file_id}", status_code=HTTPStatus.NO_CONTENT)
 async def delete_file(
     file: File = Depends(validated_file), session: Session = Depends(gen_session)
 ):
     """Delete a specific file by its id."""
-    file_location = Path(file.path)
-    file_location.unlink(missing_ok=True)
+    stmt = (
+        select(func.count())
+        .select_from(File)
+        .filter_by(project_id=file.project_id)
+        .filter_by(hash=file.hash)
+    )
```

Comment:
This should filter local files (using status) and then reuse the hash to find the local filename. The `path` hack is meant to disapear.

Reply:
I don't quite understand here. How could I avoid the `path` hacker here.

Reply:
The local filepath is constructed by concatenating the transient storage path with the hash. Since you have the hash, you can just rebuild the local file path

Reply:
I'm still not quite sure what the `path` hack refers to here and why we should rebuild the local file path from the hash.
Is this right?
```python
    stmt = (
        select(func.count())
        .select_from(File)
        .filter_by(project_id=file.project_id)
        .filter_by(hash=file.hash)
        .filter_by(status=FileSaveLocation.LOCAL.value)
    )
    number_of_duplicate_files = session.scalars(stmt).one()
    if number_of_duplicate_files == 1:
        file_location = Path(f"{file.project_id}-{file.hash}")
        file_location.unlink(missing_ok=True)
    session.delete(file)
```